feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/options/SwapOptionsViewModel.kt

```diff
@@ -10,84 +10,87 @@ import io.novafoundation.nova.common.utils.Percent
 import io.novafoundation.nova.common.utils.flowOfAll
 import io.novafoundation.nova.common.utils.formatting.format
 import io.novafoundation.nova.common.utils.formatting.formatWithoutSymbol
-import io.novafoundation.nova.common.validation.InputValidationMixinFactory
-import io.novafoundation.nova.common.validation.ValidationStatus
-import io.novafoundation.nova.common.validation.isError
-import io.novafoundation.nova.common.validation.isWarning
-import io.novafoundation.nova.common.validation.notValidOrNull
+import io.novafoundation.nova.common.validation.FieldValidationResult
 import io.novafoundation.nova.feature_swap_api.presentation.state.SwapSettings
 import io.novafoundation.nova.feature_swap_api.presentation.state.SwapSettingsStateProvider
 import io.novafoundation.nova.feature_swap_impl.R
-import io.novafoundation.nova.feature_swap_impl.domain.slippage.SlippageRepository
-import io.novafoundation.nova.feature_swap_impl.domain.validation.slippage.SlippageValidationFailure
-import io.novafoundation.nova.feature_swap_impl.domain.validation.slippage.SlippageValidationPayload
+import io.novafoundation.nova.feature_swap_impl.data.assetExchange.assetConversion.AssetConversionExchangeFactory
 import io.novafoundation.nova.feature_swap_impl.presentation.SwapRouter
+import io.novafoundation.nova.feature_swap_impl.presentation.fieldValidation.SlippageFieldValidatorFactory
 import kotlinx.coroutines.async
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.filterNotNull
 import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.flatMapLatest
 import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapNotNull
 import kotlinx.coroutines.launch
 
 class SwapOptionsViewModel(
     private val swapRouter: SwapRouter,
     private val resourceManager: ResourceManager,
     private val swapSettingsStateProvider: SwapSettingsStateProvider,
-    private val inputValidationMixinFactory: InputValidationMixinFactory<SlippageValidationPayload, SlippageValidationFailure>,
-    private val slippageRepository: SlippageRepository
+    private val assetExchangeFactory: AssetConversionExchangeFactory,
+    private val slippageFieldValidatorFactory: SlippageFieldValidatorFactory
 ) : BaseViewModel() {
 
-    private val defaultSlippage = slippageRepository.getDefaultSlippage()
-    private val slippageTips = slippageRepository.getSlippageTips()
-
     private val swapSettingState = async {
         swapSettingsStateProvider.getSwapSettingsState(viewModelScope)
     }
 
     private val swapSettingsStateFlow = flowOfAll { swapSettingState.await().selectedOption }
         .shareInBackground()
 
+    private val assetExchange = swapSettingsStateFlow.mapNotNull { it.assetIn }
+        .map { assetExchangeFactory.create(it.chainId) }
+        .filterNotNull()
+        .shareInBackground()
```

Comment:
`AssetExchange` interfacge shouldnt be exposed to presentationlayer, since it is in `data` layer. And it shouldnt be created via factory since lifecycle of `AssetExchange` instances is managed by `SwapService`. You can propagate config calls via `SwapService` and `SwapInteractor` by passing `chain` to retrieve relevant exchange in `swapService` similar to other methods in `SwapService`